server/src/main/java/io/druid/client/indexing/IndexingServiceClient.java

```diff
@@ -79,6 +88,45 @@ public void mergeSegments(List<DataSegment> segments)
     runQuery(new ClientAppendQuery(dataSource, segments));
   }
 
+  public String hadoopMergeSegments(
+      String dataSource,
+      List<Interval> intervalsToReindex,
+      AggregatorFactory[] aggregators,
+      DruidCoordinatorHadoopMergeConfig hadoopMergeConfig
+  )
+  {
+    final InputStream queryResponse = runQuery(
+        new ClientHadoopIndexQuery(
+            String.format(
```

Comment:
Because at anytime time, for each datasource there could be only 1 hadoop merge task running, otherwise HadoopMerger will not submit the reindex task. I believe DateTime() is enough to make the task ID unique.